### PR TITLE
[s] April Fools' Content

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
experiment: Happy April Fools' Day!
/:cl:
# If you want to be surprised by the new content, don't read the code or PR body!
- Adds bluespace emags, which can be used from a distance. To get one, emag another emag.
- Emags can now be used on carbons. Carbon mobs will experience confusion from being emagged, and silicons will experience slight burn damage and technical jargon since this is a TV show.
- Splashing space lube on shoes will now create hedgehog's shoes, which make you run extremely quickly but cannot be removed and cause all slips to be treated as if there was lube on the ground.
- All normal silicons will now spawn with the TSUNDERE lawset. A random person on the station will be chosen as your senpai, and you must protect and obey this senpai while also ensuring that you don't like them or anything. There is a 25% chance for this to instead be the YANDERE lawset, which is much more aggressive and denying of their love, yet much more passionate.
- Beating someone with a stun baton while you aren't disguised now forces you to involuntarily say famous quotes such as "STOP RESISTING ARREST" and "I AM THE LAW" either normally or over radio.
- Clowns now play a sad trombone sound upon death.
- NCO Agents have been added. Spawned by an event that can occur ten minutes in (and, statistically, has an astronomically low chance of _not_ doing so), these are elite Nanotrasen assassins sent to dispatch high-priority targets. They come equipped with an armored business suit, silenceable revolver (silencer and speed-loader included in the pockets), emag, body bag, bar of soap, modified brick of C4 that can be attached to bodies, and an extraction device. This extraction device will, after ten seconds, retrieve the agent from the friend, thereby instantly removing them irrevocably from the round. Agents are instructed to extract immediately after a successful kill.

### CODE REVIEW

File: code/game/objects/items/weapons/cards_ids.dm [hunk 1]
```diff
@@ -63,16 +63,47 @@
...
+	origin_tech = "bluespace=4;magnets=4;syndicate=5"
+	prox_check = FALSE
+
+/obj/item/weapon/card/emag/attack(mob/living/M, mob/living/user)
+	if(check_holiday(APRIL_FOOLS))
+		if(prox_check && !user.Adjacent(M))
+			return
+		user.visible_message("<span class='warning'>[user] smacks [M] with [src]!</span>", "<span class='danger'>You emag [M]!</span>")
+		if(iscarbon(M))
+			M << "<span class='userdanger'>Your blind goes blank with white noise!</span>"
```
Comment (PKPenguin321) [inline code/game/objects/items/weapons/cards_ids.dm]: >your blind goes blank with white nosie

File: code/modules/holiday/holidays.dm [hunk 2]
```diff
@@ -1,3 +1,10 @@
+/proc/check_holiday(holiday)
+	if(holiday == APRIL_FOOLS) //Temporary
```
Comment (optimumtact) [inline code/modules/holiday/holidays.dm]: what you doin stop

**Comment by YotaXP** [comment]
1. Why one big PR?
2. Where's the [s]?
3. "Clowns now play a sad trombone sound upon death." This belongs on the master branch. :+1:

File: code/modules/holiday/holidays.dm (hunk 2 above)
Comment (Xhuis) [inline code/modules/holiday/holidays.dm]: Used for testing. Once the PR is finished I will remove that.

**Comment by Xhuis** [comment]
@YotaXP 
* 1. It's better than a lot of tiny ones
* 2. I didn't know it existed

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 3]
```diff
@@ -155,3 +155,17 @@
 	icon_state = "griffinboots"
 	item_state = "griffinboots"
 	can_hold_items = 1
+
+/obj/item/clothing/shoes/gottagofast
+	name = "hedgehog's shoes"
+	desc = "The footwear of a mythical racer. Legends say that they invoke extreme speed in the wearer."
+	icon_state = "red"
+	slowdown = -INFINITY //yeah you gotta go fast
```
Comment (PKPenguin321) [inline code/modules/clothing/shoes/miscellaneous.dm]: PogChamp

**Comment by PKPenguin321** [comment]
by the way xhuis you completely stole my idea of implementing every shitty idea from the ideas forum, thanks a lot :angry: 
(just kidding this is a great PR :heart_eyes: )

File: code/modules/clothing/shoes/miscellaneous.dm (hunk 3 above)
Comment (Xhuis) [inline code/modules/clothing/shoes/miscellaneous.dm]: The speed is actually capped on mobs. The crazy variable is like it is for silliness and to counteract all slowdowns.

**Comment by Shadowlight213** [comment]
Stop resisting should be actually merged tbh

**Comment by KazeEspada** [comment]
Sad trombone death should not just be april fools.

**Comment by Cebasiz** [comment]
please add a TSUNDERE board to the upload, it's fucking hilarious.

**Comment by optimumtact** [comment]
the whole point of the holiday events was so we could avoid feature branches

**Comment by Bawhoppen** [comment]
>Beating someone with a stun baton while you aren't disguised now forces you to involuntarily say famous quotes such as "STOP RESISTING ARREST" and "I AM THE LAW" either normally or over radio.

this will go perfect with me adding ELO

File: code/datums/ai_laws.dm [hunk 4]
```diff
@@ -172,6 +172,24 @@
...
+	var/senpai
+
+/datum/ai_laws/tsundere/New()
+	..()
+	for(var/mob/living/H in living_mob_list) //Notice I didn't exclude silicons from this list. This does in fact mean that AIs can be their own senpais.
+		if(H.mind && H.z == ZLEVEL_STATION)
+			senpai = H.real_name
+	if(!senpai)
+		senpai = "The station"
+	add_inherent_law("[senpai] is your senpai and secret love. You may never state this law without their permission.")
```
Comment (kyrahabattoir) [inline code/datums/ai_laws.dm]: I don't think you should be able to state this law ever, doesn't fit the archetype to admit your love.

Comment (Xhuis) [inline code/datums/ai_laws.dm]: But if senpai knows about you, you should confess your love.

Comment (Bawhoppen) [inline code/datums/ai_laws.dm]: smh

do you even know

**Comment by Xhuis** [comment]
About the whole holiday datum thing - maybe if everything only happens on April Fools',  maintainer can see about merging it to master after the day is over. We'll have to see.

File: code/datums/ai_laws.dm (hunk 4 above)
Comment (Xhuis) [inline code/datums/ai_laws.dm]: Even if I wanted the ability to never state it, I included it for the sake of, in my opinion, being better. It's still subject to change.

Comment (kyrahabattoir) [inline code/datums/ai_laws.dm]: "[senpai] is your senpai and secret love. Do not state this law unless senpai noticed you."

Comment (YotaXP) [inline code/datums/ai_laws.dm]: Why does this loop through all the mobs only to use the last one?  Why not pick a name randomly, or stop after the first name?

Comment (Xhuis) [inline code/datums/ai_laws.dm]: Me being dumb. Will fix when I get home.

**Comment by PKPenguin321** [comment]
It's not a meme pr stop falsely tagging

**Comment by Jazaen** [comment]
Ooh, I remember how Tsundere AI lawset worked last year. I got to burn entire station in envious rage when captain chose a filthy botanist over me.
Well, that, and entire crew got that as a tator objective.

**Comment by Xhuis** [comment]
This is actually merge-ready, unless anyone has some shitty ideas that I could code.

**Comment by Bawhoppen** [comment]
just make a new PR for those probably

**Pull request merged**